utils/manage_events.py

```diff
@@ -0,0 +1,103 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+
+import logging
+from typing import List
+
+from cortx.utils.message_bus import MessageBus, MessageConsumer
+from ha.core.event_manager.event_manager import EventManager
+
+COMPONENT_ID = 'hare'
+
+class EventListener():
+    """This is listener for event manager
+
+    Typical use case of this class is as follows
+    subscribe(event list) - subscribe to events for listening, returns topic
+                            which can be used for listening
+    _listen(topic) - reading messages received on above topic
+    unsubscribe(event list) - unsubscribe as we dont want to receive these
+                              events anymore
+    get_message_type() - returns topic just in case user is not maintaining
+                         output of 'subscribe'
+    Note: 1. There is only 1 topic(message type, queue) for given component.
+          2. When component subcribes for event for the 1st time topic is
+             created and returned to user.
+          3. This topic will be used by user to listen for events.
+          4. After event is received in queue, user will read it, process it
+             and acknowledge it
+
+    """
+
+    event_manager = None
+    consumer = None
+
+    def __init__(self):
+        logging.debug('Inside EventListener')
+        self.event_manager = EventManager.get_instance()
+
+    def subscribe(self, event_list: List[str]) -> str:
+        logging.info('Subscribing for events: (%s)', str(event_list)[1:-1])
+        kafka_topic_name = self.event_manager.subscribe(COMPONENT_ID, event_list)
+        return kafka_topic_name
+
+    # Gets message type name mapped with component name
+    # Event manager creates message type (queue/topic) when component
+    # subscribes the event for the first time
+    def get_message_type(self) -> str:
+        logging.debug('get_message_type')
+        return self.event_manager.message_type(COMPONENT_ID)
+
+    # Get list of events subscribed by Hare
+    def get_events(self) -> List[str]:
+        logging.debug('Get list')
+        return self.event_manager.get_events(COMPONENT_ID)
+
+    def unsubscribe(self, event_list: List[str]) -> bool:
+        logging.info('Unsubscribing for events: (%s)', str(event_list)[1:-1])
+        return self.event_manager.unsubscribe(COMPONENT_ID, event_list)
+
+    # Listen for events from event manager
+    # Once user gets message using below command user needs to call ack() to
+    # acknowledge that message is already processed
+    def listen(self, topic: str, time_out):
```

Comment:
According to the code, the method doesn't listen but just takes at least one message. And doesn't return it BTW.

Normally, I would expect that  `listen` method is something blocking (so it will continuously fetch the data and hand it to the callback). In the way it is implemented a more descriptive name would be something like `get_next_message() -> Optional[MessageType]` (it returns not more than 1 message).

BTW what will happen if the user will invoke this methods twice in a row but will not invoke `ack()`? Will it return one and the same value?

Reply:
Yes, this function will return message but we still don't know what will be type(format) of message.
We are implementing skeleton that's why I haven't added return message.
During implementation of [EOS-22224](https://jts.seagate.com/browse/EOS-22224) I am planning to add proper decode and return logic.

`BTW what will happen if the user will invoke this methods twice in a row but will not invoke ack()? Will it return one and the same value?
`
That depends on event manager implementation and I was planning to add more details about it while implementing above task. 
Sounds good?

Reply:
> Yes, this function will return message but we still don't know what will be type(format) of message.

I see, but then it's a good reason to force MessageBus creators (or cortx-ha team?) to provide this information 😉 

Suggestion:
1. Add a stub class that would store the parsed message - e.g. `BaseEvent`. Nowadays it can be empty but it already can have FIXME comment. That class can have some fields in the future and it also can be inherited (e.g. different subclasses can represent different message types that our application can work with).
2. Implement a stub parsing method that takes a raw message string and returns `BaseEvent`. Nowadays its implementation can be no-op or something dummy. There should be just FIXME or TODO comment.

Why this is worth doing:
1. This approach defines the contracts (because we'll use types instead of raw strings; other parts of application will depend on those types and can freely access the right keys without any fear to get None or KeyError or something). When (by any reason) the format of the messages changes, that change will be reflected at the level of type. Other parts of our application will be validated with mypy and thus if we access some non-existent fields (the message structure changed), then we'll get build error. So the mistake will not be hidden.
2. Our parsing method, stub type and TODO/FIXME comments define our intention - so the way how this code should evolve will be defined. So although there are many things to be clarified at MessageBus or EventManager sides, we know how this code should look like in the future. Even if the code will be polished by another developer.

cc @vaibhavparatwar, @mssawant, @suvratjoshi

Reply:
@knekrasov  Changes name to 'get_next_message()', this function will return 1 message at a time.